### PR TITLE
Add documentation for ReflectionProbe

### PR DESCRIPTION
**Pull request by SamuelDeboni**

### CODE REVIEW

File: doc/classes/ReflectionProbe.xml [hunk 1]
```diff
@@ -1,8 +1,10 @@
 <?xml version="1.0" encoding="UTF-8" ?>
 <class name="ReflectionProbe" inherits="VisualInstance" category="Core" version="3.2">
 	<brief_description>
+            Capture it's surroundings to crate reflection.
```
Comment (Calinou) [inline doc/classes/ReflectionProbe.xml]:
```suggestion
			Captures its surroundings to create reflections.
```

File: doc/classes/ReflectionProbe.xml [hunk 2]
```diff
@@ -1,8 +1,10 @@
...
 	</brief_description>
-	<description>
+        <description>
+            Capture it's surroundings as a sort of 360 degrees image, and stores versions of it with increasing levels of blur to simulate diferent roughness.
```
Comment (Calinou) [inline doc/classes/ReflectionProbe.xml]:
```suggestion
			Capture its surroundings as a dual parabolid image, and stores versions of it with increasing levels of blur to simulate different material roughnesses.
```

File: doc/classes/ReflectionProbe.xml [hunk 3]
```diff
@@ -11,14 +13,17 @@
 	</methods>
 	<members>
 		<member name="box_projection" type="bool" setter="set_enable_box_projection" getter="is_box_projection_enabled" default="false">
+                    If true enable box projection.
```
Comment (Calinou) [inline doc/classes/ReflectionProbe.xml]:
```suggestion
			If [code]true[/code], enables box projection. This makes reflections look more correct in rectangle-shaped rooms by offsetting the reflection center depending on where the camera is located.
```

File: doc/classes/ReflectionProbe.xml [hunk 4]
```diff
@@ -11,14 +13,17 @@
...
 		</member>
 		<member name="cull_mask" type="int" setter="set_cull_mask" getter="get_cull_mask" default="1048575">
 		</member>
 		<member name="enable_shadows" type="bool" setter="set_enable_shadows" getter="are_shadows_enabled" default="false">
+                    If true enable shadows.
```
Comment (Calinou) [inline doc/classes/ReflectionProbe.xml]:
```suggestion
			If [code]true[/code], computes shadows in the reflection probe. This makes the reflection probe slower to render; you may want to disable this if using the [constant UPDATE_ALWAYS] [member update_mode].
```

File: doc/classes/ReflectionProbe.xml [hunk 5]
```diff
@@ -11,14 +13,17 @@
...
 		</member>
 		<member name="extents" type="Vector3" setter="set_extents" getter="get_extents" default="Vector3( 1, 1, 1 )">
 		</member>
 		<member name="intensity" type="float" setter="set_intensity" getter="get_intensity" default="1.0">
+                    Defines the reflection intensity.
```
Comment (Calinou) [inline doc/classes/ReflectionProbe.xml]:
```suggestion
			Defines the reflection intensity.
```

**Review by Calinou — CHANGES_REQUESTED** [review]
I added some suggestions to make the documentation more detailed :slightly_smiling_face:

Also, the documentation XML must be indented with tabs, not spaces (with the exception of code samples).

File: doc/classes/ReflectionProbe.xml [hunk 6]
```diff
@@ -1,8 +1,10 @@
 <?xml version="1.0" encoding="UTF-8" ?>
 <class name="ReflectionProbe" inherits="VisualInstance" category="Core" version="3.2">
 	<brief_description>
+			Captures its surroundings to create reflections.
```
Comment (akien-mga) [inline doc/classes/ReflectionProbe.xml]: Too much indentation.

File: doc/classes/ReflectionProbe.xml [hunk 7]
```diff
@@ -1,8 +1,10 @@
...
 	</brief_description>
-	<description>
+        <description>
```
Comment (akien-mga) [inline doc/classes/ReflectionProbe.xml]: Should be indented with one tab, not spaces.

File: doc/classes/ReflectionProbe.xml [hunk 8]
```diff
@@ -1,8 +1,10 @@
...
+			Capture its surroundings as a dual parabolid image, and stores versions of it with increasing levels of blur to simulate different material roughnesses.
```
Comment (akien-mga) [inline doc/classes/ReflectionProbe.xml]: Too much indentation.

**Comment by akien-mga** [comment]
There are still indentation issues, and the commits should be squashed into one. Otherwise the added docs seem good :)

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**